sdk/eventhub/event-hubs/samples-dev/sendBufferedEvents.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT Licence.
+
+/**
+ * @summary Demonstrates how to send events to an Event Hub from a buffered producer. This
+ * sample is different from the `sendEvent` one in that it uses the buffered producer
+ * client which buffers events before sending them right away. See the client constructor
+ * options for more information on how to configure the buffering behavior.
+ *
+ * @azsdk-weight 60
+ */
+
+import { AmqpAnnotatedMessage } from "@azure/core-amqp";
+import {
+  EventData,
+  EventHubBufferedProducerClient,
+  OnSendEventsErrorContext,
+} from "@azure/event-hubs";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const connectionString = process.env["EVENTHUB_CONNECTION_STRING"] || "";
+
+async function handleError(ctx: OnSendEventsErrorContext): Promise<void> {
+  console.log(`The following error occurred:`);
+  console.log(JSON.stringify(ctx.error), undefined, "  ");
+  console.log(
+    `And the following events were not sent as a result to the partition with ID ${ctx.partitionId}:`
+  );
+  console.log(
+    ctx.events.map((event: EventData | AmqpAnnotatedMessage) => JSON.stringify(event)).join("\n\n")
+  );
+}
+
+export async function main(): Promise<void> {
+  console.log(`Running sendBufferedEvents sample`);
+
+  const client = new EventHubBufferedProducerClient(connectionString, {
+    /** An error handler must be provided */
+    onSendEventsErrorHandler: handleError,
+
+    /** wait for 750 milliseconds before sending a batch */
+    maxWaitTimeInMs: 750,
+
+    /** buffer up to 1000 events per partition before sending */
+    maxEventBufferLengthPerPartition: 1000,
```

Comment:
Interesting... Since both options are configured, which one gets respected?

Reply:
Looks that if the time is up and no new event arrives, we will send the batch. otherwise, if the batch is full, we send it.

Reply:
So, whichever satisfies first? Is there a default value for both of these?

Reply:
Default for `maxWaitTimeInMs` is 1000 and default for `maxEventBufferLengthPerPartition` is 1500.

Reply:
I feel it would be useful to document that. How about this

```ts
// Create a buffered client that batches the enqueued events and sends it either
// after 750ms or after batching 1000 events, whichever occurs first.
const client = new EventHubBufferedProducerClient(connectionString, {
   onSendEventsErrorHandler: handleError,
   maxWaitTimeInMs: 750,
   maxEventBufferLengthPerPartition: 1000

Reply:
done in https://github.com/Azure/azure-sdk-for-js/pull/18674/commits/b7db4e3f6e13ceebdd6e7327ae31f3958b41c854